models.py

```diff
@@ -55,3 +55,19 @@ class Product(Model):
 product_pydantic = pydantic_model_creator(Product, name="Product")
 product_pydanticIn = pydantic_model_creator(
     Product, name="ProductIn", exclude=("percentage_discount", "id", "product_image", "date_published"))
+
+
+
+
+class Cart(Model):
+    id = fields.IntField(pk=True, index=True)
+    user = fields.ForeignKeyField("models.User", related_name="cart")
+    created_at = fields.DatetimeField(default=datetime.utcnow)
+    updated_at = fields.DatetimeField(default=datetime.utcnow)
+
+class CartItem(Model):
+    id = fields.IntField(pk=True, index=True)
+    cart = fields.ForeignKeyField("models.Cart", related_name="items")
+    product = fields.ForeignKeyField("models.Product", related_name="cart_items")
+    quantity = fields.IntField(default=1)
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding quantity validation in CartItem</b></div>
<div id="fix">

Consider adding validation for the `quantity` field in the `CartItem` model to ensure it's always positive. You might want to use `fields.IntField(default=1, gt=0)` to enforce this constraint.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -72,1 +72,1 @@
 -    quantity = fields.IntField(default=1)
 +    quantity = fields.IntField(default=1, gt=0)
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/trulyfurqan/eshop-fastapi/pull/8#issuecomment-2472732780>#b8ff8d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged